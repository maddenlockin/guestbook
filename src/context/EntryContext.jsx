import { createContext } from "react";

const EntryContext = createContext()

