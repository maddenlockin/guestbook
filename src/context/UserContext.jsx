import React, { createContext, useContext, useState,  useMemo } from "react";
//create context
const UserContext = createContext()
//create provider
const UserProvider = ({ children }) => {
  const [user, setUser] = useState('')

  const value = useMemo(() => ({ user, setUser }), [user])
  return <UserContext.Provider value={value}>
    {children}
  </UserContext.Provider>
}
//create custom hook
const useUser = () => {
  const context = useContext(UserContext)

  if (context === undefined) {
    throw new Error('useUser can only be used within the UserContext Provider');
  }
  return context
}

export{ UserProvider, useUser }