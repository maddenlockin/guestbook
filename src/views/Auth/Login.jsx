import React, { useState } from 'react'
import { useLocation, useHistory } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';
import { useForm } from '../../hooks/useForm';

export default function Login() {

  const history = useHistory();
  const location = useLocation();
  const auth = useAuth();
  const { formState, handleFormChange } = useForm({ email: '', password: '' });
  const [error, setError] = useState(null);

  const { from } = location.state || { from: { pathname: '/' } };

  const handleLogin = (e) => {
    e.preventDefault();
    const loginSuccess = auth.login(formState.email, formState.password);

    if (loginSuccess) {
      history.replace(from);
    } else {
      setError('Invalid credentials');
    }
  };
  
  return (
    <>
      <form onSubmit={handleLogin} className='login-form' >
        <label htmlFor='email'>Email</label>
        <input
          required
          id='email'
          name='email'
          type='email'
          value={formState.email}
          onChange={handleFormChange}
        />{' '}
        <label htmlFor='password'>Password</label>
        <input
          required
          id='password'
          name='password'
          type='password'
          value={formState.password}
          onChange={handleFormChange}
        />{' '}
        <button type='submit' aria-label='Sign In'>
          Sign in
        </button>
      </form>
      {error && <h2>{error}</h2>}
    </>
  );
}
