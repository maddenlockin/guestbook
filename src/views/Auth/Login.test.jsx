import React from 'react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { ProvideAuth } from '../../context/ProvideAuth';
import { UserProvider } from '../../context/UserContext';
import Login from './Login';
import { render } from '@testing-library/react';

it('should render Login view', async () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <ProvideAuth>
          <Login />
        </ProvideAuth>
      </UserProvider>
    </MemoryRouter>
  )

  expect(container).toMatchSnapshot()
})