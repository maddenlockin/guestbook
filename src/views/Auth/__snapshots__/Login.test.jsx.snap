// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login view 1`] = `
<div>
  <form
    class="login-form"
  >
    <label
      for="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      required=""
      type="email"
      value=""
    />
     
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      required=""
      type="password"
      value=""
    />
     
    <button
      aria-label="Sign In"
      type="submit"
    >
      Sign in
    </button>
  </form>
</div>
`;
